app/src/main/java/com/moviefiend/torian/moviefiend/StarView.java

```diff
@@ -0,0 +1,94 @@
+package com.moviefiend.torian.moviefiend;
+
+import android.content.Context;
+import android.graphics.Canvas;
+import android.graphics.Color;
+import android.graphics.Paint;
+import android.graphics.Path;
+import android.graphics.Region;
+import android.support.annotation.ColorInt;
+import android.support.annotation.FloatRange;
+import android.util.AttributeSet;
+import android.view.View;
+
+public class StarView extends View {
+
+    private final Paint mLinePaint;
+    private final Paint mFillPaint;
+    private final Path mStarLinePath;
+    private final Path mStarFillPath;
+
+    private @FloatRange(from=0.0, to=1.0) float mPercent;
+
+    public StarView(Context context, AttributeSet attrs) {
+        super(context, attrs);
+
+        mLinePaint = new Paint();
+        mLinePaint.setStrokeWidth(10);
+        mLinePaint.setStyle(Paint.Style.STROKE);
+        mLinePaint.setColor(Color.BLACK);
+
+        mFillPaint = new Paint();
+        mFillPaint.setStyle(Paint.Style.FILL);
+        mFillPaint.setColor(Color.YELLOW);
+
+        mStarLinePath = new Path();
+        mStarFillPath = new Path();
+
+        mPercent = 0f;
+    }
+
+    @Override
+    protected void onSizeChanged(int w, int h, int oldw, int oldh) {
+        super.onSizeChanged(w, h, oldw, oldh);
+
+        createStarPath(mStarLinePath);
+        createStarPath(mStarFillPath);
+    }
+
+    @Override
+    protected void onDraw(Canvas canvas) {
+        super.onDraw(canvas);
+
+        canvas.clipPath(mStarLinePath, Region.Op.DIFFERENCE);
+        canvas.drawPath(mStarLinePath, mLinePaint);
+
+        canvas.clipPath(mStarFillPath, Region.Op.UNION);
+        canvas.clipRect(getWidth() * mPercent, 0f, getWidth(), getHeight(), Region.Op.DIFFERENCE);
+        canvas.drawPath(mStarFillPath, mFillPaint);
+    }
+
+    public void setLineColor(@ColorInt int lineColor) {
+        mLinePaint.setColor(lineColor);
+        invalidate();
+        requestLayout();
```

Comment:
Why do you need to call `requestLayout` when you change the line color?
`invalidate` should suffice.

Calling `requestLayout` triggers `measure` and `layout` while calling `invalidate` only triggers `onDraw`.